### ISSUE TITLE
Native query results from Hibernate are binding as null to dto fields

### ISSUE THREAD

**Issue by suryamegham1729** [micronaut-projects/micronaut-data#1138]
### Expected Behavior

The query is expected to run without errors and return the result.

### Actual Behaviour

A DataAccessException is thrown with the following message: Null argument specified for [field].

### Steps To Reproduce

1. Create a CrudRepository interface.
2. Annotate a method with @Query setting nativeQuery=true and making sure that the query columns are matching with the constructor parameters of dto.
3. Call the repository method from your application or a unit test.

### Environment Information

- Operating System: macOS 10.15.7
- JDK version: AdoptOpenJDK 13.0.2

### Example Application

fc00760660db7124cf1877a1c69430aea79636de

### Version

2.5.0

**Comment by dstepanov** [comment]
Did it work before?

**Comment by suryamegham1729** [comment]
yes @dstepanov , have provided the [fix](https://github.com/micronaut-projects/micronaut-data/pull/1137) now, please have a look

**Comment by suryamegham1729** [comment]
hi @dstepanov and @graemerocher , have raised the PR with the right account, please have a look
